### PR TITLE
Add type definition for tcp-ping.

### PR DESCRIPTION
**Pull request by stegano**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by stegano** [comment]
How do I select a checklist of comments from dt-bot?

File: types/tcp-ping/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                  port: number,
+                  attempts: number,
+                  avg: number,
+                  max: number,
+                  min: number,
+                  results: {
+                      seq: number,
+                      time: number,
+                      error: Error
+                  }
```
Comment (plantain-00) [inline types/tcp-ping/index.d.ts]: The type of `results` is array, `time` and `error` can be `undefined`
https://github.com/apaszke/tcp-ping/blob/master/ping.js#L65
https://github.com/apaszke/tcp-ping/blob/master/ping.js#L71

File: types/tcp-ping/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// Type definitions for tcp-ping 0.1
+// Project: https://github.com/wesolyromek/tcp-ping/issues
```
Comment (plantain-00) [inline types/tcp-ping/index.d.ts]: It should be the project url, rather than the issue url.

File: types/tcp-ping/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                   port: number,
+                   attempts: number,
+                   avg: number,
+                   max: number,
+                   min: number,
+                   results: {
+                       seq: number,
+                       time: number,
+                       error: Error
+                   }
```
Comment (plantain-00) [inline types/tcp-ping/index.d.ts]: Same here, you can define an interface to reuse them like:
```js
interface Result = { address: string; ... }
export function ping(options: Option, callback: (error: Error, result: Result) => void): void;
export function probe(address: string, port: number, callback: (error: Error, result: Result) => void): void;
```

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Comment by plantain-00** [comment]
Don't worry about the checklist of comments from dt-bot, it is expected to be selected by owners or maintainers, or not expected to be selected.

**Comment by stegano** [comment]
@plantain-00 Thanks for letting me know. :)

File: types/tcp-ping/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export interface Result {
+    address: string;
+    port: number;
+    attempts: number;
+    avg: number;
+    max: number;
+    min: number;
+    results: {
```
Comment (plantain-00) [inline types/tcp-ping/index.d.ts]: The type of `results` is array

Comment (stegano) [reply types/tcp-ping/index.d.ts]: Thank you! :)

**Review by plantain-00 — APPROVED** [review]

**Pull request merged**